drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -404,32 +422,35 @@ impl<S: SpiServer> Io<S> {
                                 ringbuf_entry!(Trace::ErrRespNoPayload);
                                 continue;
                             }
-                            errcode =
-                                SprotError::from(rxmsg.payload_error_byte());
-                            ringbuf_entry!(Trace::SprotError(errcode));
-                            if is_recoverable_error(errcode) {
-                                // TODO: There are rare cases where
-                                // the RoT dose not receive
-                                // a 0x01 as the first byte in a message.
-                                // See issue #929.
+                            let err =
+                                SprotError::from_u8(rxmsg.payload_error_byte())
+                                    .unwrap_or(SprotError::Unknown);
```

Comment:
This plumbing is very confusing to me.

We take a `u8`, decode it into `Option<SprotError>`, unwrap that into either the value or `SprotError::Unknown`, then wrap it in `Some(...)` again.

It seems like we're tracking two mostly-separate things:

- The most recent error – which may or may not exist – and is returned at the end of the function, at which point it is guaranteed to exist (since we're unwrapping it).
- Whether we successfully deserialized the error byte

Combine them in this way is confusing.  The nicely-typed option would be to add a new wrapper type, e.g.
```rust
enum SpSprotError {
    Sprot(SprotError),
    BadDeserialization
}
```

However, this couldn't be used in Idol right now, because it doesn't support complex error types.

The dumb way to do it would be
```rust
enum SpSprotError {
    NoMessage = 1,
    BadTransferSize,
    TaskRestart,
    InvalidCrc,
    FlowError,
    // literally every other field from SprotError
    DeserializationFailed, // <-- this is new
}

impl From<SprotError> for SpSprotError {
    fn from(s: SprotError) -> Self {
        match s {
            SprotError::NoMessage => Self::NoMessage,
            // etc
        }
    }
}
```

Any other ideas on how to clean up this section?